web/Controllers/PropertyController.cs

```diff
@@ -140,250 +140,286 @@ public ActionResult GetPublicViewLink(int propertyId)
+        [FromForm] PropertyInputModelTenantPreferences model,
+        [FromRoute] int propertyId, [FromRoute] string operationType)
     {
-        var property = _propertyService.GetPropertyByPropertyId(propertyId);
-        if (property == null)
-        {
-            return StatusCode(404);
-        }
+        const string nextActionName = "PropertyInputStepSixAvailability";
 
-        var newPropertyModel = PropertyViewModel.FromDbModel(property);
-        newPropertyModel.PropertyId = propertyId;
+        return await StandardPropertyInputPostMethod(model, propertyId, operationType, nextActionName);
+    }
 
-        if (!(GetCurrentUser().IsAdmin || GetCurrentUser().LandlordId == newPropertyModel.LandlordId))
+    [Authorize(Roles = "Landlord, Admin")]
+    [HttpGet("{propertyId:int}/{operationType:regex(^(add|edit)$)}/step/6")]
+    public ActionResult PropertyInputStepSixAvailability([FromRoute] int propertyId, [FromRoute] string operationType)
+    {
+        const string viewName = "PropertyInputForm/Availability";
+        var model = new PropertyInputModelAvailability
         {
-            return StatusCode(403);
-        }
+            Step = 6,
+            IsEdit = OperationTypeToIsEdit(operationType)
+        };
 
-        return View("EditProperty", new AddNewPropertyViewModel { Step = step, Property = newPropertyModel });
+        return StandardPropertyInputGetMethod(model, propertyId, viewName);
     }
 
     [Authorize(Roles = "Landlord, Admin")]
-    [HttpPost("edit/step/{step:int}")]
-    public async Task<IActionResult> EditProperty_Continue([FromRoute] int step, int propertyId,
-        [FromForm] PropertyViewModel newPropertyModel, int? landlordId = null)
+    [HttpPost("{propertyId:int}/{operationType:regex(^(add|edit)$)}/step/6")]
+    public async Task<ActionResult> PropertyInputStepSixAvailability([FromForm] PropertyInputModelAvailability model,
+        [FromRoute] int propertyId, [FromRoute] string operationType)
     {
-        newPropertyModel.PropertyId = propertyId;
-
         if (!ModelState.IsValid)
         {
-            return View("EditProperty", new AddNewPropertyViewModel { Step = step, Property = newPropertyModel });
+            return RedirectToAction("PropertyInputStepSixAvailability", "Property",
+                new
+                {
+                    propertyId,
+                    operationType
+                });
+        }
+
+        if (!_propertyService.IsUserAdminOrCorrectLandlord(CurrentUser, propertyId))
+        {
+            return StatusCode(403);
         }
 
+        var propertyView = model.FormToViewModel();
+
+        await _propertyService.UpdateProperty(propertyId, propertyView);
+
+        return RedirectToAction("ViewProperty", "Property", new { propertyId });
+    }
+
+    [Authorize(Roles = "Landlord, Admin")]
+    [HttpGet(
+        "/landlord/{landlordId:int}/property/{operationType:regex(^(add|edit)$)})/{propertyId:int}/cancel")]

```

Comment:
is the operation type needed in the route?

Reply:
No, but I kept it here because thought the route had to stay consistent, so for this endpoint the rest would stay the same while cancel would be added. I will remove it :)